src/odemis/driver/symphotime.py

```diff
@@ -0,0 +1,891 @@
+            self._socket = socket.create_connection((self._host, self._port))
+            self._socket.settimeout(2.0)
+
+        except socket.error:
+            raise model.HwError("Failed to connect to '%s:%d', check that the Symphotime "
+                                "Server is connected to the network, turned "
+                                "on, and correctly configured." % (host, port))
+
+        # to acquire before sending anything on the socket
+        self._net_access = threading.Lock()
+        
+        # try get parameters from metadata
+        self._metadata[model.MD_PIXEL_SIZE] = (10e-6, 10e-6)
+        self._metadata[model.MD_DESCRIPTION] = ""
+        
+        # Internal paramters
+        self._measurement_type = PQ_MEASTYPE_IMAGESCAN
+        
+        # Vigilant attributes
+        try:
+            ckwargs = children["scanner"]
+        except KeyError:
+            raise ValueError("No 'scanner' child configuration provided")
+            
+        self.scanner = Scanner(parent=self, **ckwargs)
+        self.children.value.add(self.scanner)
+
+        # Measurement parameters
+        self.data = BasicDataFlow(self.StartMeasurement, self.StopMeasurement)
+        self._tStart = None  # this will hold the epoch time a measurmenet starts
+
+        # Create a threadEmitter to listen for incoming commands
+        self._shutdown_flag = threading.Event()
+        self._user_break = threading.Event()
+        self._measuring_flag = BEvent()
+        self._listener_thread = threading.Thread(target=self._listen)
+        self._listener_thread.daemon = True 
+        self._listener_thread.start()
+
+    def terminate(self):
+        if self.isMeasuring():
+            self.StopMeasurement()
+        self.waitTillMeasurementComplete()
+        # Wait a bit in case additional messages should be received.
+        time.sleep(2.0)
+        # if we don't do this, the server won't disconnect nicely
+        # we don't want to wait indefinitely though, in case there are no messages.
+        self._shutdown_flag.set()
+        self._listener_thread.join()  # wait till the listener thread closes
+        self._socket.shutdown(socket.SHUT_RDWR)
+        self._socket.close()
+
+    def isMeasuring(self):
+        '''
+        Returns true if an acquisition is running. False otherwise.
+        '''
+        return self._measuring_flag.is_set()
+
+    def waitTillMeasurementStarts(self):
+        self._measuring_flag.wait_true()
+
+    def waitTillMeasurementComplete(self):
+        logging.debug("Waiting for measurmenet to complete...")
+        self._measuring_flag.wait_false()
+        logging.debug("Measurement complete. Done waiting. ")
+
+    def _sendMessage(self, msg):
+        '''
+        Sends an order to the device.
+        msg: Message type object
+        '''
+        logging.debug("S: Msgtype %s", msg)
+        time.sleep(0.02)  # wait a bit to prevent flooding the server.
+        self._net_access.acquire()
```

Comment:
Use `with self._net_access:`. The danger of explicitly calling `.acquire()` and `.release()` is that if the code in between fails, the lock will never be released.